flink-libraries/flink-table/src/main/scala/org/apache/flink/table/functions/utils/UserDefinedFunctionUtils.scala

```diff
@@ -382,7 +381,10 @@ object UserDefinedFunctionUtils {
       implicitResultType: TypeInformation[_],
       params: Expression*): TableFunctionCall = {
     val arguments = transformLiteralExpressions(params: _*)
-    val userDefinedResultType = tableFunction.getResultType(arguments)
+    val typeInformations = params.map { param =>
+      if (param.valid) param.resultType else null
```

Comment:
I think if the parameter cannot be validated, we do not need to continue here because translation would later fail.